### PR TITLE
Store to global in more paths

### PR DESCRIPTION
**Pull request by lekko-jonathan**

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -1,139 +1,169 @@
...
+        const res = await this.client.getFloatValue(req);
+        return res.value;
     }
-    return {};
-  }
 
-  async getProto(namespace: string, key: string, ctx?: ClientContext): Promise<Any> {
-    if (!ctx) {
-      ctx = new ClientContext();
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (passichenko) [inline src/client.ts:95]: looks like you need to move it line 99

File: src/index.ts [hunk 2]
```diff
@@ -24,7 +24,8 @@ function initAPIClient(options: APIOptions): AsyncClient {
       protocol: options.transportProtocol ?? TransportProtocol.HTTP,
       apiKey: options.apiKey
     }).build();
-  return new TransportClient(options.repositoryOwner, options.repositoryName, transport);
+  const client =  new TransportClient(options.repositoryOwner, options.repositoryName, transport);
+  return client;
```
Comment (passichenko) [inline src/index.ts:28]: why this change?

**Review by passichenko — APPROVED** [review]

**Comment by passichenko** [comment]
looks like you also changed how context can be passed, can you mention that for posterity?

**Comment by lekko-jonathan** [comment]
The context passing change is to make it so that we can generate code that doesn't need to know which sdk it is using.  Right now, we have to import context into the generated code to do fromJson.. this just makes it cleaner
